packages/moonstone/ProgressBar/ProgressBarTooltip.js

```diff
@@ -11,22 +12,84 @@ import Tooltip from '../TooltipDecorator/Tooltip';
 
 import css from './ProgressBarTooltip.module.less';
 
+// prop-type validator that warns on invalid orientation + position
+/* istanbul ignore next */
+const validatePosition = (base) => (props, key, componentName, location, propFullName, ...rest) => {
+	const {position} = props;
+	let result = base(props, key, componentName, location, propFullName, ...rest);
+
+	if (!result && position) {
+		const orientation = props.orientation || 'horizontal';
+		const hasVerticalValue = ['before', 'after', 'left', 'right'].includes(position);
+		if (
+			(orientation === 'vertical' && !hasVerticalValue) ||
+			(orientation === 'horizontal' && hasVerticalValue)
+		) {
+			result = new Error(
+				`'${key}' value '${position}' is not a valid value for the orientation '${orientation}'`
+			);
+		}
+	}
+
+	return result;
+};
+
 const memoizedPercentFormatter = memoize((/* locale */) => new NumFmt({
 	type: 'percentage',
 	useNative: false
 }));
 
-const getSide = (orientation, side) => {
-	const valid = orientation === 'vertical' || (
-		orientation === 'horizontal' && (side === 'before' || side === 'after')
-	);
+const getDefaultPosition = (orientation) => orientation === 'horizontal' ? 'above' : 'before';
 
-	warning(
-		valid,
-		'The value of `side` must be either "after" or "before" when `orientation` is "horizontal"'
-	);
+// Returns an array of keywords with horizontal first and vetical second
+const getSide = (orientation, side, position) => {
+	if (position || !side) {
+		position = position || getDefaultPosition(orientation);
 
-	return valid ? side : 'before';
+		if (orientation === 'horizontal') {
+			switch (position) {
+				case 'above':
+				case 'below':
+					return ['auto', position];
+				case 'after':
+				case 'before':
+				case 'left':
+				case 'right':
```

Comment:
As I mentioned before, I super-prefer to see the defaults in `default:`.  I guess there are more than twice as many valid ones as these 4 invalid ones.  Could possibly get all the 'above' and 'below' options on one line each?

Reply:
okay